app/assets/javascripts/discourse/app/components/sidebar/anonymous/custom-sections.js

```diff
@@ -1,16 +1,22 @@
 import Component from "@glimmer/component";
 import { inject as service } from "@ember/service";
 import Section from "discourse/components/sidebar/user/section";
+import CommunitySection from "discourse/components/sidebar/common/community-section";
 
 export default class SidebarAnonymousCustomSections extends Component {
   @service router;
   @service site;
+  @service siteSettings;
 
   get sections() {
     return this.site.anonymous_sidebar_sections?.map((section) => {
-      return new Section({
+      const klass = section.section_type ? CommunitySection : Section;
```

Comment:
I think we can do some anticipating of future changes here and probably use a switch case statement here.